src/components/Tabs/index.js

```diff
@@ -0,0 +1,223 @@
+    display: block;
+    position: absolute;
+    border-bottom: 1px solid ${themes.tm.gray04};
+    left: 0;
+    right: 0;
+    bottom: 0;
+    z-index: ${zIndex.unset};
+  }
+
+  &.overflowed:after {
+    width: 100vw;
+    left: calc(50% - 50vw);
+  }
+`;
+
+const Content = styled.div.attrs({
+  className: props => (props.areTabsOverflow ? "overflowed" : null)
+})`
+  display: flex;
+  flex: 1;
+  width: 100%;
+  overflow-x: auto;
+  scroll-behavior: smooth;
+
+  &.overflowed {
+    margin-left: calc(50% - 50vw);
+    padding-left: calc(50vw - 50%);
+    width: 100vw;
+  }
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+`;
+
+const Item = styled.div`
+  ${largeAndUp`
+    padding-right: ${spacing.comfy};
+  `};
+  display: block;
+  padding-right: ${spacing.normal};
+  white-space: nowrap;
+
+  &:last-of-type {
+    padding-right: 0;
+  }
+`;
+
+const checkIfOverflows = ({ offsetWidth = 0, scrollWidth = 0 } = {}) =>
+  offsetWidth < scrollWidth;
+
+const TabItemButton = styled.div.attrs({
+  className: props => (props.isActive ? "isActive" : ""),
+  "data-index": props => props.dataIndex
+})`
+  display: block;
+  padding: ${padding} 0 ${spacing.cozy} 0;
+  cursor: pointer;
+  &.isActive {
+    border-bottom: 4px solid
+      ${props => props.underlineColor || colors.azure.base};
+  }
+`;
+
+class Tabs extends Component {
+  static propTypes = {
+    items: PropTypes.arrayOf(PropTypes.node).isRequired,
+    index: PropTypes.number,
+    headingIndex: PropTypes.number,
+    onClick: PropTypes.func,
+    variant: PropTypes.string,
+    accent: PropTypes.string,
+    weight: PropTypes.string,
+    underlineColor: PropTypes.string
+  };
+
+  static defaultProps = {
+    index: -1,
+    headingIndex: -1,
+    onClick: () => {},
+    variant: null,
+    accent: null,
+    weight: null,
+    underlineColor: null
+  };
+
+  state = {
+    isTabsContainerOverflows: false
+  };
+
+  componentDidMount() {
+    this.updateIsOverflowsState();
+    window.addEventListener("resize", this.updateIsOverflowsState); // eslint-disable-line
+  }
+
+  componentDidUpdate() {
+    if (this.state.isTabsContainerOverflows) {
+      const {
+        offsetWidth: contentWidth,
+        scrollLeft: contentScrollLeft
+      } = this.content;
+      const { index } = this.props;
+      const {
+        offsetWidth: activeItemWidth = 0,
+        offsetLeft: activeItemLeft = 0
+      } = this.itemRefs[index];
+
+      if (activeItemWidth + activeItemLeft > contentWidth + contentScrollLeft) {
+        this.content.scrollLeft =
+          activeItemWidth +
+          activeItemLeft -
+          contentWidth +
+          parseInt(padding, 10);
+      } else if (activeItemLeft < contentScrollLeft) {
+        this.content.scrollLeft = activeItemLeft - parseInt(padding, 10);
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateIsOverflowsState); //eslint-disable-line
+  }
+
+  updateIsOverflowsState = () => {
```

Comment:
Can you please explain why we need to account for this overflow state? I would like to handle it with a css-only approach, if possible.

Reply:
If tabs container is overflowed, it takes the whole width of the screen (ignoring parent's padding) on small and medium screens. Also, in this case tabs container will scroll to the selected tab, if it is out of visible part of container.